src/services/db/GitFileCommitService.ts

```diff
@@ -101,51 +113,43 @@ export default class GitFileCommitService {
     const defaultBranch = STAGING_BRANCH
     logger.info("Updating file in local Git file system")
     const filePath = directoryName ? `${directoryName}/${fileName}` : fileName
-    const updatePromises = [
-      this.gitFileSystemService.update(
-        sessionData.siteName,
-        filePath,
-        fileContent,
-        sha,
-        sessionData.isomerUserId,
-        defaultBranch
-      ),
-    ]
+    const stagingUpdateResult = await this.gitFileSystemService.update(
+      sessionData.siteName,
+      filePath,
+      fileContent,
+      sha,
+      sessionData.isomerUserId,
+      defaultBranch
+    )
 
     const shouldUpdateStagingLite =
-      filePath &&
+      !!filePath &&
       isReduceBuildTimesWhitelistedRepo(sessionData.growthbook) &&
       !isFileAsset({ fileName, directoryName })
 
+    let stagingLiteUpdateResult
     if (shouldUpdateStagingLite) {
-      updatePromises.push(
-        this.gitFileSystemService.update(
-          sessionData.siteName,
-          filePath,
-          fileContent,
-          sha,
-          sessionData.isomerUserId,
-          this.STAGING_LITE_BRANCH
-        )
+      stagingLiteUpdateResult = await this.gitFileSystemService.update(
+        sessionData.siteName,
+        filePath,
+        fileContent,
+        sha,
+        sessionData.isomerUserId,
+        STAGING_LITE_BRANCH
       )
     }
 
-    const results = await Promise.all(updatePromises)
-    const [stagingUpdateResult, stagingLiteUpdateResult] = results
-
     if (stagingUpdateResult.isErr()) {
       throw stagingUpdateResult.error
-    } else if (shouldUpdateStagingLite && stagingLiteUpdateResult.isErr()) {
+    } else if (
+      shouldUpdateStagingLite &&
+      stagingLiteUpdateResult &&
+      stagingLiteUpdateResult.isErr()
+    ) {
       throw stagingLiteUpdateResult.error
     }
 
-    this.gitFileSystemService.push(sessionData.siteName, defaultBranch)
-    if (shouldUpdateStagingLite) {
-      this.gitFileSystemService.push(
-        sessionData.siteName,
-        this.STAGING_LITE_BRANCH
-      )
-    }
+    this.pushToGithub(sessionData, shouldUpdateStagingLite)
```

Comment:
same issue as above; there's also a repeated pattern - we could probably just separate out the steps as functions and chain them together.